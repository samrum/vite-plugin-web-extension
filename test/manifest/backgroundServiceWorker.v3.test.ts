import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("backgroundServiceWorker");

runManifestV3Test("backgroundServiceWorker", () => ({
  background: {
    service_worker: `${resourceDir}/serviceWorker.js`,
  },
}));
