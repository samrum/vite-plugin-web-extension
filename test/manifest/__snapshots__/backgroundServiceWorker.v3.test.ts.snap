// Vitest Snapshot v1

exports[`backgroundServiceWorker - Manifest V3 1`] = `
[
  {
    "code": "console.log(\\"serviceWorker\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/backgroundServiceWorker/serviceWorker.js",
    "fileName": "assets/test/manifest/resources/backgroundServiceWorker/serviceWorker.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/backgroundServiceWorker/serviceWorker.js": {
        "code": "console.log(\\"serviceWorker\\");",
        "originalLength": 30,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 29,
      },
    },
    "name": "test/manifest/resources/backgroundServiceWorker/serviceWorker",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"background\\": {
    \\"service_worker\\": \\"serviceWorker.js\\",
    \\"type\\": \\"module\\"
  }
}",
    "type": "asset",
  },
  {
    "fileName": "serviceWorker.js",
    "name": undefined,
    "source": "import \\"/assets/test/manifest/resources/backgroundServiceWorker/serviceWorker.js\\";",
    "type": "asset",
  },
]
`;
